mlapi/discriminating_algo.py

```diff
@@ -96,9 +99,23 @@ def execute_discriminating_facets_algorithm(self, unique_documents_by_facet):
         uniformly_distributed_facets = self.get_uniformly_distributed_facets(facets_sample)
         return self.get_facets_with_max_values(uniformly_distributed_facets)
 
-    def sort_discriminating_facets(self, unique_documents_by_facet):
+    def score_discriminating_facets(self, unique_documents_by_facet):
         values_by_facet = self.get_values_by_facet(unique_documents_by_facet)
         documents_count_by_facet_name = self.get_documents_count_by_facet_name(unique_documents_by_facet)
-        return sorted(values_by_facet.items(),
-                      key=lambda values_by_facet : len(values_by_facet[1]) * documents_count_by_facet_name[values_by_facet[0]],
-                      reverse= True)
\ No newline at end of file
+
+        max_value = 0
+        min_value = 100
+        eps = 1e-16
+        for (facetName, facetValue) in values_by_facet.items():
+            value = len(facetValue) * documents_count_by_facet_name[facetName]
+            if value > max_value:
+                max_value = value
+            if value < min_value:
+                min_value = value
+        difference_max_min = max_value - min_value + eps
+        facet_values_score = []
+        for (facetName, facetValue) in values_by_facet.items():
+            score = (len(facetValue) * documents_count_by_facet_name[facetName] - min_value) / difference_max_min
+            facet_values_score.append(FacetScoreValues(facetName, facetValue, score))
+        facet_values_score.sort(key=lambda facet: facet.score, reverse=True)
+        return facet_values_score
```

Comment:
À tester, mais je pense que tu pourrais t'en sauver avec quelquechose du genre :

facet_values_score = [FacetScoreValues(facetName, facetValue, (len(facetValue) * documents_count_by_facet_name[facetName])) for (facetName, facetValue) in values_by_facet.items()]
max_value = max([facetScoreValue.score for facetScoreValue in facet_values_score])
for facetScoreValue in facet_values_score:
	facetScoreValue = facetScoreValue / max_value